src/common/error_test.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ *    Copyright (c) 2020, The OpenThread Commissioner Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file defines test cases for Error class.
+ */
+
+#include "commissioner/error.hpp"
+
+#include <catch2/catch.hpp>
+
+namespace ot {
+
+namespace commissioner {
+
+TEST_CASE("error-message", "[error]")
+{
+    SECTION("default Error object has no error message")
+    {
+        Error error;
+
+        REQUIRE(error.GetCode() == ErrorCode::kNone);
+        REQUIRE(error.GetMessage().empty());
+    }
+}
+
+TEST_CASE("error-to-string", "[error]")
+{
+    SECTION("default Error object should return 'NONE' as string representation")
+    {
+        Error error;
+
+        REQUIRE(error.ToString() == "NONE");
```

Comment:
- Use `OK` just like openthread?

Reply:
> Test string formatting for unsuccessful erros

Only `NONE` is documented for the `Error::ToString` API and string representation of other error are not documented and may change in the future. That means the user can reply on the truth that `Error(kNone).ToString()` always returns `NONE`, but no guarantee is provided for the rest errors.

> Test that error can be compared with error code
This has been tested by `TEST_CASE("error-comparison", "[error]")` at below.

> Test the behavior of comparing unsuccessful errors
I am not sure if we need to do, I am assuming that if comparison of one kind of error works, then the rest will work. Do we need to test if `ErrorA1 != ErrorB1`? There will be `17 * 17` testcases for this, and I think it is not necessary.

Reply:
ok ot me.